core/time/timezone/tz_unix.odin

```diff
@@ -0,0 +1,91 @@
+#+build darwin, linux, freebsd, openbsd, netbsd
+package timezone
+
+import "core:os"
+import "core:strings"
+import "core:path/filepath"
+import "core:time/datetime"
+
+local_tz_name :: proc(allocator := context.allocator) -> (name: string, success: bool) {
+	local_str, ok := os.lookup_env("TZ", allocator)
+	if !ok {
+		orig_localtime_path := "/etc/localtime"
+		path, err := os.absolute_path_from_relative(orig_localtime_path)
+		if err != nil {
+			// If we can't find /etc/localtime, fallback to UTC
+			if err == .ENOENT {
+				return strings.clone("UTC", allocator), true
+			}
+
+			return
+		}
+		defer delete(path)
+
+		// FreeBSD makes me sad.
+		if path == orig_localtime_path {
+			data, ok2 := os.read_entire_file_from_filename("/var/db/zoneinfo")
+			if !ok2 {
+				return
+			}
+			defer delete(data)
+
+			tzone_name := strings.clone(strings.trim_space(string(data)), allocator)
```

Comment:
Could you get away with a `trim_right_space` only? That would allow you to return the original file read allocation, not sure if there could be spaces on the left though.